app/javascript/components/util/DeleteDialog.js

```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import { PropTypes } from 'prop-types';
+import { Button, Form, Modal } from 'react-bootstrap';
+
+class DeleteDialog extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      loading: false,
+      disabled: true,
+      showTextInput: false,
+    };
+  }
+
+  handleReasonChange = event => {
+    event.persist();
+    const showTextInput = event.target.value === 'Other';
+    this.setState({ disabled: false, showTextInput, [event.target.id]: event.target.value }, () => {
+      this.props.onChange(event);
+    });
+  };
+
+  delete = () => {
+    this.setState({ loading: true }, () => {
+      this.props.delete();
+    });
+  };
+
+  render() {
+    return (
+      <Modal size="lg" show centered onHide={this.props.toggle}>
+        <Modal.Header>
+          <Modal.Title>Delete {this.props.type}</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <p>
+            Are you sure you want to delete this {this.props.type}? This action cannot be undone. For auditing purposes, this deletion will be available in this
+            record&apos;s history export.
+          </p>
+          <p>Please select reason for deletion:</p>
+          <Form.Control as="select" className="form-control-md mb-3" id="delete_reason" onChange={this.handleReasonChange} defaultValue={-1}>
+            <option disabled value={-1}>
+              --
+            </option>
+            <option>Duplicate entry</option>
+            <option>Entered in error</option>
+            <option>Other</option>
+          </Form.Control>
+          {this.state.showTextInput && (
+            <Form.Control
```

Comment:
this needs a character limit (which also needs to be validated in the back end). the following screenshot was taken from the histories edit/delete PR, but same idea/problem

![Screen Shot 2021-05-05 at 2 54 17 AM](https://user-images.githubusercontent.com/9956561/117112230-bfb47700-ad56-11eb-9d2e-0b3b93e1138d.png)

Reply:
@sgober wrote the delete dialog modal and since it's being made in parallel we're gonna have to coordinate any changes we make to it. Tagging her so we can discuss

Reply:
@ashankland - I can do this and send the updated file to you and @mayerm94 (I have some minor local tweaks anyway). This would validated it on the front end.  How many characters do you think @timwongj?

Reply:
We have 200 character limits in a few other front-end fields so maybe that would be a good place to start?

Reply:
@holmesie what do you think?

Reply:
yeah, for delete_reason 200 characters is perfectly reasonable.